godot-core/src/builtin/vector2.rs

```diff
@@ -85,6 +81,206 @@ impl Vector2 {
+    ) -> Self {
+        let x = cubic_interpolate_in_time(
+            self.x, b.x, pre_a.x, post_b.x, weight, b_t, pre_a_t, post_b_t,
+        );
+        let y = cubic_interpolate_in_time(
+            self.y, b.y, pre_a.y, post_b.y, weight, b_t, pre_a_t, post_b_t,
+        );
+
+        Self::new(x, y)
+    }
+
+    pub fn direction_to(self, to: Self) -> Self {
+        (to - self).normalized()
+    }
+
+    pub fn distance_squared_to(self, to: Self) -> f32 {
+        self.to_glam().distance_squared(to.to_glam())
+    }
+
+    pub fn distance_to(self, to: Self) -> f32 {
+        self.to_glam().distance(to.to_glam())
+    }
+
+    pub fn dot(self, other: Self) -> f32 {
+        self.to_glam().dot(other.to_glam())
+    }
+
+    pub fn floor(self) -> Self {
+        Self::from_glam(self.to_glam().floor())
+    }
+
+    pub fn from_angle(angle: f32) -> Self {
+        Self::from_glam(glam::Vec2::from_angle(angle))
+    }
+
+    pub fn is_equal_approx(self, to: Self) -> bool {
+        is_equal_approx(self.x, to.x) && is_equal_approx(self.y, to.y)
+    }
+
+    pub fn is_finite(self) -> bool {
+        self.to_glam().is_finite()
+    }
+
+    pub fn is_normalized(self) -> bool {
+        self.to_glam().is_normalized()
+    }
+
+    pub fn is_zero_approx(self) -> bool {
+        is_zero_approx(self.x) && is_zero_approx(self.y)
+    }
+
+    pub fn length_squared(self) -> f32 {
+        self.to_glam().length_squared()
+    }
+
+    pub fn lerp(self, to: Self, weight: f32) -> Self {
+        Self::from_glam(self.to_glam().lerp(to.to_glam(), weight))
+    }
+
+    pub fn limit_length(self, length: Option<f32>) -> Self {
+        Self::from_glam(self.to_glam().clamp_length_max(length.unwrap_or(1.0)))
+    }
+
+    pub fn max_axis_index(self) -> i32 {
+        let me = self.to_glam().max_element();
+        (me == self.y) as i32
+    }
+
+    pub fn min_axis_index(self) -> i32 {
+        let me = self.to_glam().min_element();
+        (me == self.y) as i32
+    }
+
+    pub fn move_toward(self, to: Self, delta: f32) -> Self {
+        let vd = to - self;
+        let len = vd.length();
+        if len <= delta || len < CMP_EPSILON {
+            to
+        } else {
+            self + vd / len * delta
+        }
+    }
+
+    pub fn orthogonal(self) -> Self {
+        Self::new(self.y, -self.x)
+    }
+
+    pub fn posmod(self, pmod: f32) -> Self {
+        Self::new(fposmod(self.x, pmod), fposmod(self.y, pmod))
+    }
+
+    pub fn posmodv(self, modv: Self) -> Self {
+        Self::new(fposmod(self.x, modv.x), fposmod(self.y, modv.y))
+    }
+
+    pub fn project(self, b: Self) -> Self {
+        Self::from_glam(self.to_glam().project_onto(b.to_glam()))
+    }
+
+    pub fn reflect(self, normal: Self) -> Self {
+        Self::from_glam(self.to_glam().reject_from(normal.to_glam()))
+    }
+
+    pub fn round(self) -> Self {
+        Self::from_glam(self.to_glam().round())
+    }
+
+    pub fn sign(self) -> Self {
+        -self
+    }
```

Comment:
This is not the sign...

Reply:
apologies i should have looked closer. was origonally under the guise that it was a sign flip, not extracting 1 or -1